file2.js

```diff
@@ -0,0 +1,558 @@
+      $(".btnGenerateKey").click(function() {
+        $(this).addClass('disabled');
+        var url = $(this).data("url");
+        utils.post(url, null, function(e) {
+          if (e.error) {
+            $(".GenerateKeyFeedback").showError(e.error);
+          } else {
+            location.reload();
+          }
+        }, function() {
+          $(".GenerateKeyFeedback").showError("Could not generate key, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindGenerateUserKey = function() {
+      $(".btnGenerateUserKey").click(function() {
+        $(this).addClass('disabled');
+        var url = $(this).data("url");
+        utils.post(url, null, function(e) {
+          if (e.error) {
+            $(".GenerateUserKeyFeedback").showError(e.error);
+          } else {
+            location.reload();
+          }
+        }, function() {
+          $(".GenerateUserKeyFeedback").showError("Could not add key to user, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindRemoveProject = function() {
+      $(".btnRemoveProject").click(function() {
+        var $this = $(this);
+        var removeUrl = $this.data("url");
+        var redirectUrl = $this.data("iden");
+
+
+        utils.post(removeUrl, {}, function(e) {
+          if (e.error) {
+            $(".RemoveProjectFeedback").showError(e.error);
+          } else {
+            window.location.href = redirectUrl;
+          }
+        }, function() {
+          $(".RemoveProjectFeedback").showError("Could not remove project, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindTransferProject = function() {
+
+      var $orgBox = $("select.js-project-destination");
+      var $teamBox = $("select.js-organization-teams-selection-box");
+
+      $(".btnTransferProject").click(function() {
+        var $this = $(this);
+
+        var payload = {};
+
+        payload.orgId = $orgBox.val();
+
+        var teamIds = $teamBox.val();
+
+        if (teamIds) {
+          payload.teamIds = teamIds;
+          var url = $this.data("url");
+          utils.JSend(url, payload, function() {
+              $(".TransferProjectFeedback").showSuccess("Project transferred successfully!");
+              window.location = $this.data("reload");
+            }, function(error) {
+              $(".TransferProjectFeedback").showError(error);
+            },
+            "Could not transfer project, please try again.");
+        }
+      });
+
+      //Reflects the teams that are shown on the team selection box
+      var teamBoxData = [];
+
+      //Bindings on each item on the organization selection box
+      $orgBox.on("rendered.bs.select",function() {
+        var teamsUrl = $orgBox.find("option:selected").data("url");
+        $teamBox.prop("disabled", true).selectpicker("refresh");
+        utils.request(teamsUrl, successFallback, failureFallback, function() {
```

Comment:
![Codacy](https://app.dev.codacy.org/assets/images/favicon.png) Issue found: ['failureFallback' was used before it was defined. (no-use-before-define)](https://app.dev.codacy.org/app/codacytestuser/CheckPrCommentsPublicProject/pullRequest?prid=7992)